flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/config/FlinkConfigManager.java

```diff
@@ -0,0 +1,183 @@
+                        .build(
+                                new CacheLoader<>() {
+                                    @Override
+                                    public Configuration load(
+                                            Tuple4<Long, String, String, ObjectNode> k) {
+                                        return generateConfig(k);
+                                    }
+                                });
+
+        updateDefaultConfig(defaultConfig);
+        if (defaultConfig.getBoolean(OPERATOR_DYNAMIC_CONFIG_ENABLED)) {
+            scheduleConfigWatcher();
+        }
+    }
+
+    public Configuration getDefaultConfig() {
+        return defaultConfig;
+    }
+
+    @VisibleForTesting
+    public void updateDefaultConfig(Configuration newConf) {
+        if (newConf.equals(defaultConfig)) {
+            LOG.info("Default configuration did not change, nothing to do...");
+            return;
+        }
+
+        LOG.info("Updating default configuration");
+        this.operatorConfiguration =
+                FlinkOperatorConfiguration.fromConfiguration(newConf, namespaces);
+        this.defaultConfig = newConf.clone();
+        // We do not invalidate the cache to avoid deleting currently used temp files, simply bump
+        // the version
+        this.defaultConfigVersion.incrementAndGet();
+    }
+
+    public FlinkOperatorConfiguration getOperatorConfiguration() {
+        return operatorConfiguration;
+    }
+
+    public Configuration getDeployConfig(ObjectMeta objectMeta, FlinkDeploymentSpec spec) {
+        return getConfig(objectMeta, spec);
+    }
+
+    public Configuration getObserveConfig(FlinkDeployment deployment) {
+        return getConfig(deployment.getMetadata(), ReconciliationUtils.getDeployedSpec(deployment));
+    }
+
+    @SneakyThrows
+    private Configuration getConfig(ObjectMeta objectMeta, FlinkDeploymentSpec spec) {
+        var key =
+                Tuple4.of(
+                        defaultConfigVersion.get(),
+                        objectMeta.getNamespace(),
+                        objectMeta.getName(),
+                        objectMapper.convertValue(spec, ObjectNode.class));
+
+        // Always return a copy of the configuration to avoid polluting the cache
+        return cache.get(key).clone();
+    }
+
+    private Configuration generateConfig(Tuple4<Long, String, String, ObjectNode> key) {
+        try {
+            LOG.info("Generating new config");
+            return FlinkConfigBuilder.buildFrom(
+                    key.f1,
+                    key.f2,
+                    objectMapper.convertValue(key.f3, FlinkDeploymentSpec.class),
+                    defaultConfig);
+        } catch (Exception e) {
+            throw new RuntimeException("Failed to load configuration", e);
+        }
+    }
+
+    private void scheduleConfigWatcher() {
+        var checkInterval = defaultConfig.get(OPERATOR_DYNAMIC_CONFIG_CHECK_INTERVAL);
```

Comment:
nit: we can directly use `defaultConfig.get(OPERATOR_DYNAMIC_CONFIG_CHECK_INTERVAL).toMillis();`